quinn-proto/src/crypto/rustls.rs

```diff
@@ -256,6 +261,9 @@ pub struct HandshakeData {
     ///
     /// Always `None` for outgoing connections
     pub server_name: Option<String>,
+
+    /// The ciphersuite negotiated with the peer.
+    pub negotiated_cipher_suite: CipherSuite,
```

Comment:
Yeah, but on the other hand why not? :D

This is `rustls.rs` module, which already depends on `rustls` specific (and `quinn` re-export `rustls` as well)

Reply:
But if we're going to have a rustls-specific type, why not `SupportedCipherSuite` instead of `CipherSuite`?

@Ralith have opinions on this?

Reply:
`SupportedCipherSuite` also contains the implementation.

I thought the scope of this `crypto::rustls::HandshakeData` for `quinn` is just give the information about the negotiated cipher and implementation is not needed (reducing overhead of the struct); that's why I went for `CipherSuite` instead.

Anyway lemme know what do you prefer

Reply:
Sorry, this slipped through the cracks!

I don't see a meaningful difference in cost between `SupportedCipherSuite` and `CipherSuite`. The latter is a C-style enum, and the former is an enum bearing static references; both are small `Copy` values, and none of this is on a hot path anyway.

If we were to go the `SupportedCipherSuite` direction, I'd be tempted to take it a step further and use `&'static Tls13CipherSuite` directly, since QUIC specifies TLS 1.3.

However, my instinct is that `CipherSuite` is easier to work with. Unlike `SupportedCipherSuite`, matching on it does not require knowledge of the specific crypto backend used, and it also has a straightforward `Debug` implementation. While @djc is more up on rustls internals than I, in the absence of a compelling argument otherwise I favor `CipherSuite`.

Reply:
It might also help to examine the intended use case of this feature. What do you want to do with this information?

Reply:
>  I favor CipherSuite

As a drive-by comment I recently changed rustls-ffi's APIs from handling `SupportedCipherSuite`s to doing something equivalent to just exposing `CipherSuite` details. I think it would be unusual for someone to want anything other than the IANA registered ID, or the "name" that corresponds to that ID and that's what the `CipherSuite` enum captures.